packages/core/src/utils/StorageUtil.ts

```diff
@@ -12,16 +12,21 @@ import type { ConnectionStatus, SocialProvider, WcWallet } from './TypeUtil.js'
 // -- Utility -----------------------------------------------------------------
 export const StorageUtil = {
   getActiveNetworkProps() {
-    const activeNamespace = StorageUtil.getActiveNamespace()
-    const activeCaipNetworkId = StorageUtil.getActiveCaipNetworkId()
-    const caipNetworkIdFromStorage = activeCaipNetworkId
-      ? activeCaipNetworkId.split(':')[1]
+    const namespace = StorageUtil.getActiveNamespace()
+    const caipNetworkId = StorageUtil.getActiveCaipNetworkId() as CaipNetworkId | undefined
+    const stringChainId = caipNetworkId ? caipNetworkId.split(':')[1] : undefined
+
+    // eslint-disable-next-line no-nested-ternary
+    const chainId = stringChainId
+      ? isNaN(Number(stringChainId))
+        ? stringChainId
+        : Number(stringChainId)
```

Comment:
how come we can have NaN here?

Reply:
When we parse `caipNetworkId`, for both EVM and other chains the `chainId` will be string, but it's EVM only when `Number(stringChainId)`

This is to return correct typed chainId instead just string for `chainId`